command/acl_auth_method_create.go

```diff
@@ -0,0 +1,174 @@
+			"-name":           complete.PredictAnything,
+			"-type":           complete.PredictAnything,
+			"-max-token-ttl":  complete.PredictAnything,
+			"-token-locality": complete.PredictAnything,
+			"-default":        complete.PredictAnything,
+			"-config":         complete.PredictNothing,
+		})
+}
+
+func (a *ACLAuthMethodCreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (a *ACLAuthMethodCreateCommand) Synopsis() string { return "Create a new ACL auth method" }
+
+// Name returns the name of this command.
+func (a *ACLAuthMethodCreateCommand) Name() string { return "acl auth-method create" }
+
+// Run satisfies the cli.Command Run function.
+func (a *ACLAuthMethodCreateCommand) Run(args []string) int {
+
+	flags := a.Meta.FlagSet(a.Name(), FlagSetClient)
+	flags.Usage = func() { a.Ui.Output(a.Help()) }
+	flags.StringVar(&a.name, "name", "", "")
+	flags.StringVar(&a.methodType, "type", "", "")
+	flags.StringVar(&a.tokenLocality, "token-locality", "local", "")
+	flags.DurationVar(&a.maxTokenTTL, "max-token-ttl", 0, "")
+	flags.BoolVar(&a.isDefault, "default", true, "")
+	flags.StringVar(&a.config, "config", "", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments.
+	if len(flags.Args()) != 0 {
+		a.Ui.Error("This command takes no arguments")
+		a.Ui.Error(commandErrorText(a))
+		return 1
+	}
+
+	// Perform some basic validation
+	if a.name == "" {
+		a.Ui.Error("ACL auth method name must be specified using the -name flag")
+		return 1
+	}
+	if !slices.Contains([]string{"global", "local"}, a.tokenLocality) {
+		a.Ui.Error("Token locality must be set to either 'local' or 'global'")
+		return 1
+	}
+	if strings.ToLower(a.methodType) != "oidc" {
+		a.Ui.Error("ACL auth method type must be set to 'OIDC'")
+		return 1
+	}
+	if len(a.config) == 0 {
+		a.Ui.Error("Must provide ACL auth method config in JSON format")
+		return 1
+	}
+
+	configJSON, err := configStringToAuthConfig(a.config)
+	if err != nil {
+		a.Ui.Error(fmt.Sprintf("Unable to parse JSON config: %v", err))
+		return 1
+	}
+
+	// Set up the auth method with the passed parameters.
+	authMethod := api.ACLAuthMethod{
+		Name:          a.name,
+		Type:          a.methodType,
+		TokenLocality: a.tokenLocality,
+		MaxTokenTTL:   a.maxTokenTTL,
+		Default:       a.isDefault,
+		Config:        configJSON,
+	}
+
+	// Get the HTTP client.
+	client, err := a.Meta.Client()
+	if err != nil {
+		a.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	// Create the auth method via the API.
+	_, err = client.ACLAuthMethods().Create(&authMethod, nil)
+	if err != nil {
+		a.Ui.Error(fmt.Sprintf("Error creating ACL auth method: %v", err))
+		return 1
+	}
+
+	a.Ui.Output(fmt.Sprintf("Created ACL auth method %s", a.name))
+	return 0
+}
+
+func configStringToAuthConfig(conf string) (*api.ACLAuthMethodConfig, error) {
```

Comment:
If we have time before release or as a followup, it would be nice to support loading files and even reading from stdin also.

Reply:
Good idea!

I added support in 234393f47d2c77d1d3e0d824ad72981b867f6ae6 and tests in 92cf8246f61f04c09e033a38376bc83abc4968bb.